crates/sargon-uniffi/src/profile/mfa/security_structures/security_shield_builder.rs

```diff
@@ -382,6 +384,28 @@ impl SecurityShieldBuilder {
         self.get(|builder| builder.validate().map(|x| x.into()))
     }
 
+    pub fn validate_role_in_isolation(
+        &self,
+        role: RoleKind,
+    ) -> Option<SecurityShieldBuilderInvalidReason> {
+        self.get(|builder| {
+            builder
+                .validate_role_in_isolation(role.into_internal())
+                .map(|x| x.into())
+        })
+    }
+
+    pub fn selected_factor_sources_for_role_status(
```

Comment:
So this is meant to be used in Select Factors for Transactions screen? If so, do we need a an instance of a builder to validate the selected combination of the factors?

Additionally, the fact that it returns `SelectedFactorSourcesForRoleStatus`, implies that the builder might contain invalid factors, but AFAIK we are actively trying to prevent this.

Reply:
> do we need an instance of a builder

Not necessarily, @CyonAlexRDX suggested this. Also, it seems easier since all the current validation functions are on a builder instance.

>  implies that the builder might contain invalid factors

Not the factors themselves, but in the context of a specific role.